sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -130,7 +163,14 @@ def __iter__(self):
     try:
       while True:
         try:
-          element = self.element_queue.get()
+          if self.element_queue.empty():
+            # We check the current step only when the Queue is empty because we
+            # will likely block waiting (if the queue is not empty, then we'll
+            # be able to get from it instantly). This allows us to limit how
+            # much we check the current step.
+            self.read_counter.check_step()
+          with self.read_counter:
```

Comment:
<!--thread_id:cc_158386091_t; commit:4255543cdc7cfcadd29633f321599d2f591021d8; resolved:1-->
<!--section:context-quote-->
> **pabloem** wrote:
> I have run two of the Python Benchmark tests - I ran each job 10 times with each configuration (baseline, change behind experiment, change activated). The results are:
> 
> sideinputs_iterable_benchmark:
> Baseline: Mean: 401s, Stdev: 13s.
> With Deactivated Change: Mean: 533s, Stdev 18s.
> With Change: Mean: 660s, Stdev: 60s
> 
> wordcount_many_files_1gb (which uses sideinputs in the write step)
> Baseline: Mean: 560s, Stdev: 40s
> With Deactivated Change: 567s, Stdev 21s.
> With Change: 548s, Stdev: 17s.
> 
> I would conclude that the change does definitely affect sideinput performance, but this does not represent a significant change for the wordcount pipeline. I can try similar benchmarks with another integration test that you may recommend.
> -P.

<!--section:body-->
I've removed the change that increased the main fetching path for side inputs. The performance impact with experiment deactivated will be zero now.